docs/docs/api.md

```diff
@@ -0,0 +1,324 @@
+---
+outline: [2, 3]
+---
+
+# PGlite API
+
+## Main Constructor
+
+`new PGlite(dataDir: string, options: PGliteOptions)`<br/>
+`new PGlite(options: PGliteOptions)`
+
+A new PGlite instance is created using the `new PGlite()` constructor.
+
+This is imported as:
+
+```ts
+import { PGlite } from "@electric-sql/pglite";
+```
+
+`await PGlite.create(dataDir: string, options: PGliteOptions)`<br />
+`await PGlite.create(options: PGliteOptions)`
+
+There is also a `PGlite.create()` static method that returns a promise, resolving to the new PGlite instance. There are a couple of advantages to using the static method:
+
+- This awaits the [`.waitReady`](#waitready) promise, ensuring that the database has fully initiated.
+- When using TypeScript and extensions, the returned PGlite instance will have the extensions namespace on its type. This is not possible with the standard constructor due to limitation with TypeScript.
```

Comment:
```suggestion
- When using TypeScript and extensions, the returned PGlite instance will have the extensions namespace on its type. This is not possible with the standard constructor due to TypesScript limitations.
```

if you have a thread/issue for the limitations perhaps we could link it here? Or not mention it at all if we only push the static factory method.